src/redux/actions/Wishlist.js

```diff
@@ -0,0 +1,150 @@
+import {
+    ADD_TO_WISHLIST_ERROR,
+    ADD_TO_WISHLIST_START,
+    ADD_TO_WISHLIST_SUCCESS,
+    VIEW_WISHLIST_ERROR,
+    VIEW_WISHLIST_START,
+    VIEW_WISHLIST_SUCCESS,
+    REMOVE_FROM_WISHLIST_ERROR,
+    REMOVE_FROM_WISHLIST_START,
+    REMOVE_FROM_WISHLIST_SUCCESS,
+  } from "../types";
+  
+  export const addToWishlist = (id) => {
+    return async (dispatch) => {
+      dispatch({
+        type: ADD_TO_WISHLIST_START,
+        payload: true,
+      });
+  
+      try {
+        const token = localStorage.getItem("token");
+        const response = await fetch(
+          `https://e-comm-team-amigos-bn-project.onrender.com/wishlist/add/${id}`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+              Authorization: `Bearer ${token}`,
+            },
+          }
+        );
+  
+        const result = await response.json();  
+        if (response.status === 200) {
+          dispatch({
+            type: ADD_TO_WISHLIST_SUCCESS,
+            payload: response.ok,
+          });
+          dispatch({
+            type: ADD_TO_WISHLIST_START,
+            payload: true,
+          });
+          dispatch({
+            type: ADD_TO_WISHLIST_START,
+            payload: false,
```

Comment:
You are dispatching many time and sometime with the same payload, what the use of sending true and false at once,

Reply:
This is beacuse when adding starts and the action ends, it should set payload to false otherwise it will be loading... endlessly even when the action was done!